ddtrace/tracer/tracer.go

```diff
@@ -306,16 +316,50 @@ func (t *tracer) worker(tick <-chan time.Time) {
 	}
 }
 
-func (t *tracer) pushTrace(trace []*span) {
+// traceInfo contains span list and sampling decision of the trace
+// needed to run single span sampling.
```

Comment:
It's better to document something as what it is or what it stands for, rather than what its fields are and where it is used. Considering you follow my suggestions, I'd do something like:
```go
// finishedTrace holds information about a trace that has finished, including its spans.
```
I would not mention that it's needed for sampling. When I read "this is needed for ..." it sounds a bit like it's an excuse for having it. As if it would not be here otherwise. Let's make it clear that it is useful instead.